devito/types/sparse.py

```diff
@@ -120,6 +140,22 @@ def __subfunc_setup__(self, key, suffix, dtype=None):
         if key is not None and not isinstance(key, SubFunction):
             key = np.array(key)
 
+        # Check if already a SubFunction
+        d = self.indices[self._sparse_position]
+        if isinstance(key, SubFunction):
+            if d in key.indices:
+                # Can use as is, dimension already matches
+                if self.alias:
```

Comment:
```
if self.alias:
    return key._rebuild(alias=self.alias, name=name)
return key
```

Reply:
I like to keep the else for simple blocks and unindent when it's a decent code chunck that is the "main" part